model/DataObject.php

```diff
@@ -1575,6 +1575,11 @@ public function getComponents($componentName, $filter = "", $sort = "", $join =
 			);
 		}
 
+		if($filter !== null || $sort !== null || $limit !== null) {
```

Comment:
You need to check join too, even though it had previously been discarded we should really not discard it now.